src/app/app.component.ts

```diff
@@ -917,6 +921,10 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
   dropToFolder(folderId): void {
     const messageIds = this.canvastable.rows.selectedMessageIds();
 
+    let currentFolderId;
+    this.messagelistservice.folderListSubject.subscribe(folders => {
```

Comment:
I think a race is possible here: messageActionsHandler should probably wait for this call to finish before it fires its own.

Reply:
folderListSubject is a BehavorSubject, which is documented as returning its current value immediately..
we already use the exact same construction in existing code ... so I think its fine?

Reply:
Moved the variable closer to its use, and switched to behaviorsubject .value, which we don't have to wait on. (If user's are running actions against messages/folders, and don't have folders, something is quite broken..)